BedrockServer.cpp

```diff
@@ -1187,8 +1189,8 @@ void BedrockServer::postPoll(fd_map& fdm, uint64_t& nextActivity) {
     }
     if (!_suppressCommandPort && (state == SQLiteNode::MASTERING || state == SQLiteNode::SLAVING) &&
         _shutdownState.load() == RUNNING) {
-        // Open the port
-        if (!_commandPort) {
+        // Open the port if we don't have one and we've had as many opens as we have closures.
+        if (!_commandPort && _commandPortMutex.try_lock()) {
```

Comment:
That's not really how `try_lock` works, create a `lock_guard` around the block of code that calls `openPort`.